packages/@sfx/core/test/unit/common/utils/dependencies.test.ts

```diff
@@ -0,0 +1,258 @@
+    it('should return an object without inherited properties', () => {
+      const merged = mergeDependencyGraphs();
+
+      expect(Object.getPrototypeOf(merged)).to.be.null;
+    });
+
+    it('should "merge" one graph', () => {
+      const first = {
+        a: ['b'],
+        b: [],
+        c: ['c'],
+      };
+
+      const merged = mergeDependencyGraphs(first);
+
+      expect(merged).to.deep.equal({
+        a: ['b'],
+        b: [],
+        c: ['c'],
+      });
+    });
+
+    it('should merge disjunct graphs', () => {
+      const first = {
+        a: ['b'],
+        b: [],
+        c: ['c'],
+      };
+      const second = {
+        d: ['e', 'f'],
+        e: [],
+        f: [],
+      };
+      const third = {};
+
+      const merged = mergeDependencyGraphs(first, second, third);
+
+      expect(merged).to.deep.equal({
+        a: ['b'],
+        b: [],
+        c: ['c'],
+        d: ['e', 'f'],
+        e: [],
+        f: [],
+      });
+    });
+
+    it('should merge the values of non-disjunct graphs', () => {
+      const first = {
+        a: ['b'],
+        b: [],
+        c: ['c'],
+      };
+      const second = {
+        a: ['d'],
+        d: ['e', 'f'],
+        e: [],
+      };
+      const third = {};
+      const fourth = {
+        a: ['f'],
+        b: [],
+        d: [],
+        f: [],
+      };
+
+      const merged = mergeDependencyGraphs(first, second, third, fourth);
+
+      expect(merged).to.deep.equal({
+        a: ['b', 'd', 'f'],
+        b: [],
+        c: ['c'],
+        d: ['e', 'f'],
+        e: [],
+        f: [],
+      });
+    });
+  });
+
+  describe('removeFromDependencyGraph()', () => {
+    it('should remove nothing given no plugin names', () => {
+      const graph = {
+        a: [],
+        b: ['a', 'c'],
+        c: [],
+      };
+
+      const newGraph = removeFromDependencyGraph(graph, []);
+
+      expect(newGraph).to.deep.equal({
+        a: [],
+        b: ['a', 'c'],
+        c: [],
+      });
+    });
+
+    it('should throw when attempting to remove a plugin that will break a dependency', () => {
+      const graph = {
+        a: ['b'],
+        b: [],
+        c: [],
+      };
+
+      const callback = () => removeFromDependencyGraph(graph, ['a']);
+
+      expect(callback).to.throw();
+    });
+
+    it('should return a new graph with the specified dependencies removed', () => {
+      const graph = {
+        a: [],
+        b: ['a', 'c'],
+        c: [],
+        d: [],
+      };
+
+      const newGraph = removeFromDependencyGraph(graph, ['a', 'd']);
+
+      expect(newGraph).to.deep.equal({
+        b: ['c'],
+        c: [],
+      });
+    });
+
+    it('should not modify the original graph', () => {
+      const dependersOfPluginA = [];
+      const graph = {
+        a: dependersOfPluginA,
+        b: ['a', 'c'],
+        c: [],
+      };
+
+      const newGraph = removeFromDependencyGraph(graph, ['a']);
+
+      expect(newGraph).not.to.equal(graph);
```

Comment:
Is this enough? Should we assert that the new graph does not have a key of `a`?

Reply:
It is, but I think it would make this test much clearer. Effectively:
```
expect oldGraph to have key 'a'
expect newGraph to not have key 'a'
```
This tells a nice little story.